### PR TITLE
[DOCS] New rule fields for prerequisites: Related Integrations, Required Fields, and Setup

### LINKED ISSUE #2015: [DOCS] Update documentation to include the new Rule fields: `Related Integrations`, `Required Fields` and `Setup`
## Description 

With https://github.com/elastic/kibana/pull/131475, https://github.com/elastic/kibana/pull/132409 and https://github.com/elastic/kibana/pull/132667, we've added three new fields (`Related Integrations`, `Required Fields` and `Setup`) to the Rule object to be initially supported in Prebuilt Rules (for 8.3), with support added for Custom Rules in the future.

With this initial implementation, these new fields will be used to show information about related integrations (and their install state) on the Rules Table and Rule Details, and other setup information on the Rule Details page like necessary ingest configuration or what fields are required for this rule to fire. This information will be displayed for any Prebuilt Rules that start including this information in 8.3.

As mentioned, these new fields are only visible with Prebuilt Rules, and so there is limited API support and currently no UI for editing them. If a Prebuilt Rule is duplicated, these fields are emptied (set to `''` or `[]`). When a Rule is exported these fields are included (as empty values), and it is possible to edit the `ndjson` and re-import and then see these fields for the Custom Rule (but still not editable in the UI). This is expected behavior, and is actually a nice and easy way to test. 

Here is a sample export you can paste into an `test.ndjson` file and import to test this feature:

```
{"id":"6cc39c80-da3a-11ec-9fce-65c1a0bee904","updated_at":"2022-05-23T01:48:23.422Z","updated_by":"elastic","created_at":"2022-05-23T01:48:20.940Z","created_by":"elastic","name":"Testing #131475, don't mind me...","tags":["Elastic","Endpoint Security"],"interval":"5m","enabled":false,"description":"Generates a detection alert each time an Elastic Endpoint Security alert is received. Enabling this rule allows you to immediately begin investigating your Endpoint alerts.","risk_score":47,"severity":"medium","license":"Elastic License v2","output_index":".siem-signals-default","meta":{"from":"5m"},"rule_name_override":"message","timestamp_override":"event.ingested","author":["Elastic"],"false_positives":[],"from":"now-600s","rule_id":"2c66bf23-6ae9-4eb2-859e-446bea181ae9","max_signals":10000,"risk_score_mapping":[{"field":"event.risk_score","operator":"equals","value":""}],"severity_mapping":[{"field":"event.severity","operator":"equals","severity":"low","value":"21"},{"field":"event.severity","operator":"equals","severity":"medium","value":"47"},{"field":"event.severity","operator":"equals","severity":"high","value":"73"},{"field":"event.severity","operator":"equals","severity":"critical","value":"99"}],"threat":[],"to":"now","references":[],"version":7,"exceptions_list":[{"id":"endpoint_list","list_id":"endpoint_list","namespace_type":"agnostic","type":"endpoint"}],"immutable":false,"related_integrations":[{"package":"system","version":"1.6.4"},{"package":"aws","integration":"cloudtrail","version":"1.11.0"}],"required_fields":[{"ecs":true,"name":"event.code","type":"keyword"},{"ecs":true,"name":"message","type":"match_only_text"},{"ecs":false,"name":"winlog.event_data.AttributeLDAPDisplayName","type":"keyword"},{"ecs":false,"name":"winlog.event_data.AttributeValue","type":"keyword"},{"ecs":false,"name":"winlog.event_data.ShareName","type":"keyword"},{"ecs":false,"name":"winlog.event_data.RelativeTargetName","type":"keyword"},{"ecs":false,"name":"winlog.event_data.AccessList","type":"keyword"}],"setup":"## Config\\n\\nThe 'Audit Detailed File Share' audit policy must be configured (Success Failure).\\nSteps to implement the logging policy with with Advanced Audit Configuration:\\n\\n```\\nComputer Configuration > \\nPolicies > \\nWindows Settings > \\nSecurity Settings > \\nAdvanced Audit Policies Configuration > \\nAudit Policies > \\nObject Access > \\nAudit Detailed File Share (Success,Failure)\\n```\\n\\nThe 'Audit Directory Service Changes' audit policy must be configured (Success Failure).\\nSteps to implement the logging policy with with Advanced Audit Configuration:\\n\\n```\\nComputer Configuration > \\nPolicies > \\nWindows Settings > \\nSecurity Settings > \\nAdvanced Audit Policies Configuration > \\nAudit Policies > \\nDS Access > \\nAudit Directory Service Changes (Success,Failure)\\n```\\n","type":"query","language":"kuery","index":["logs-endpoint.alerts-*"],"query":"event.kind:alert and event.module:(endpoint and not endgame)\\n","filters":[],"throttle":"no_actions","actions":[]}
{"exported_count":1,"exported_rules_count":1,"missing_rules":[],"missing_rules_count":0,"exported_exception_list_count":0,"exported_exception_list_item_count":0,"missing_exception_list_item_count":0,"missing_exception_list_items":[],"missing_exception_lists":[],"missing_exception_lists_count":0}
```

## Acceptance Test Criteria

* Add a documentation blurb about this new feature to the Rules Table and Rule Details
* No need to update API documentation till we add support for custom rules
* From [8.3 Detection Rules Area Checklist](https://github.com/elastic/security-team/issues/3535):
  - [ ] Associate prebuilt rules with Related Integrations:
    - [ ] New rule fields added to the REST API responses
    - [ ] UI for Related Integrations on the Rule Management page
    - [ ] UI for Related Integrations on the Rule Details page
    - [ ] UI for Required Fields on the Rule Details page
    - [ ] UI for Setup Guide on the Rule Details page
    - [ ] Kibana Advanced Setting for toggling the `Related Integrations` feature on the Rules/Monitoring tables
 

## Notes

Please see myself or @banderror for any clarification here 🙂

### PR DESCRIPTION
**Pull request by joepeeples**
Addresses #2015.

Previews:
- [Manage detection rules | View related integrations and required fields](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-management.html#rule-prerequisites) — new section
- [Advanced Settings](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/advanced-settings.html#show-related-integrations) — new section
- Detection rule APIs — added fields to response samples:
   - [Create rule](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/rules-api-create.html#_response_payload) (5 response examples)
   - [Get rule](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/rules-api-get.html#_response_payload_2)
   - [Find rules](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/rules-api-find.html#_response_payload_3)
   - [Update rule](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/rules-api-update.html#_response_payload_4)
   - Bulk rule action ([Example 1](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/bulk-actions-rules-api.html#_response_payload_8) and [Example 2](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/bulk-actions-rules-api.html#_example_payload))
- [Release notes 8.3.0](https://security-docs_2069.docs-preview.app.elstc.co/guide/en/security/master/release-notes-header-8.3.0.html#release-notes-8.3.0) — Added under "Features"

### Joe's Li'l Checklist

- [ ] Update 8.4 Release Notes and What's New once we confirm new fields & content are going into 8.4 (Separate PR because 8.4 RNs and What's New have already been merged for the release): https://github.com/elastic/security-docs/pull/2384
- [ ] ⚠️ NOTE: We also need to update What's New in 8.3, but we can't do that with this PR because it's merging to `main`, where What's New is being worked on for 8.4. I created PR #2327 to update 8.3 directly.

### CODE REVIEW

**Comment by joepeeples** [comment]
This feature isn't going into the 8.3.0 release, but keeping the `v8.3.0` label for now in case the feature becomes available soon.

**Review by jethr0null — COMMENTED** [review]
LGTM!

File: docs/detections/rules-ui-manage.asciidoc [hunk 1]
```diff
@@ -82,6 +83,24 @@ image::images/install-prebuilt-settings.png[]
 [role="screenshot"]
 image::images/install-prebuilt-rules.png[]
 
+[float]
+[[rule-prerequisites]]
+=== Confirm rule prerequisites
+
+Many Elastic prebuilt rules are designed to work with specific Elastic integrations and data fields. These prerequisites are identified in the *Related integrations* and *Required fields* fields on a rule's details page (*Detect* -> *Rules*, then click a rule's name). *Related integrations* also displays each integration's installation status and includes links for installing and configuring the listed integrations.
```
Comment (banderror) [inline docs/detections/rules-ui-manage.asciidoc:90]: The sidebar menu structure has been changed and this is not correct anymore: `*Detect* -> *Rules*`.

File: docs/release-notes/8.3.asciidoc [hunk 2]
```diff
@@ -180,6 +180,8 @@ On the next scheduled rule execution, the *Last Response* value for the rule wil
 * Provides an option to select a severity level for a case ({pull}131626[#131626]).
 * Adds the experimental *Alerts* tab to cases, which allows users to inspect attached alerts ({pull}131883[#131883]).
 * Adds the *Average time to close* metric to the Cases page ({pull}131909[#131909]).
+* Adds new fields to prebuilt detection rules' schemas: `related_integrations`, `required_fields`, and `setup` ({pull}132409[#132409]).
+* Adds *Related integrations*, *Required fields*, and *Setup guide* sections to the rule details page, to help users identify, install, and configure a rule's prerequisites. Also adds related integrations badge to the rules table ({pull}131475[#131475]). Content for these new sections is delivered in a prebuilt rules update, independent of {stack} release versioning.
```
Comment (banderror) [inline docs/release-notes/8.3.asciidoc:184]: Are we going to update the 8.3 release notes? Am I understanding correctly that we'll do it only if rules with those fields will be released in one of the `8.3.x` Out-Of-Band updates for the `Prebuilt Detection Rules` package?

File: docs/detections/rules-ui-manage.asciidoc (hunk 1 above)
Comment (joepeeples) [reply docs/detections/rules-ui-manage.asciidoc:90]:
```suggestion
Many Elastic prebuilt rules are designed to work with specific Elastic integrations and data fields. These prerequisites are identified in the *Related integrations* and *Required fields* fields on a rule's details page (*Manage* -> *Rules*, then click a rule's name). *Related integrations* also displays each integration's installation status and includes links for installing and configuring the listed integrations. 
```

Comment (banderror) [reply docs/detections/rules-ui-manage.asciidoc:90]: In 8.4 the page is hidden under this `Manage` menu:

Screenshot 2022-08-23 at 21 41 27

File: docs/release-notes/8.3.asciidoc (hunk 2 above)
Comment (joepeeples) [reply docs/release-notes/8.3.asciidoc:184]: @banderror Yes, this section will update the 8.3 release notes (the file name is version-specific). 

For 8.4 release notes, we have a [separate PR](https://github.com/elastic/security-docs/pull/2314) that's currently in progress, though it doesn't yet mention the new fields because they're not yet available in the UI. Once I can confirm that, I'll add the same wording from above to those 8.4 notes.

Basically before we merge anything related to the new fields, I'm waiting to confirm that the latest prebuilt rules package has been published and the new fields are available in both 8.3.x and 8.4.0.

File: docs/detections/rules-ui-manage.asciidoc [hunk 3]
```diff
@@ -82,6 +83,24 @@ image::images/install-prebuilt-settings.png[]
...
+
+[role="screenshot"]
+image::images/rule-details-prerequisites.png[Rule details page with Related integrations, Required fields, and Setup guide highlighted]
+
+You can also check rules' related integrations in the *Rules* and *Rule Monitoring* tables. Click the integrations badge to display the related integrations in a popup.
+
+[role="screenshot"]
+image::images/rules-table-related-integrations.png[Rules table with related integrations popup,75%]
+
+TIP: You can hide the related integrations badge in the rules tables. Go to *{kib}* -> *Stack Management* -> *Advanced Settings*, then turn off `securitySolution:showRelatedIntegrations`.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-manage.asciidoc:102]:
```suggestion
TIP: You can hide the related integrations badge in the Rules tables. Go to *{kib}* -> *Stack Management* -> *Advanced Settings*, then turn off `securitySolution:showRelatedIntegrations`.
```

File: docs/release-notes/8.3.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [inline docs/release-notes/8.3.asciidoc:184]:
```suggestion
* Adds the *Related integrations*, *Required fields*, and *Setup guide* sections to the rule details page to help users identify, install, and configure a rule's prerequisites. Also adds the related integrations badge to the Rules table ({pull}131475[#131475]). Content for these new sections is delivered in a prebuilt rules update, independent of {stack} release versioning.
```

**Review by nastasha-solomon — APPROVED** [review]
Just a couple minor edits. LGTM otherwise!

**Review by banderror — APPROVED** [review]
Looks great, thank you! 👍

File: docs/detections/rules-ui-manage.asciidoc [hunk 4]
```diff
@@ -82,6 +83,24 @@ image::images/install-prebuilt-settings.png[]
...
+
+[role="screenshot"]
+image::images/rule-details-prerequisites.png[Rule details page with Related integrations, Required fields, and Setup guide highlighted]
+
+You can also check rules' related integrations in the *Rules* and *Rule Monitoring* tables. Click the integrations badge to display the related integrations in a popup.
+
+[role="screenshot"]
+image::images/rules-table-related-integrations.png[Rules table with related integrations popup,75%]
+
+TIP: You can hide the related integrations badge in the Rules tables. Go to *{kib}* -> *Stack Management* -> *Advanced Settings*, then turn off `securitySolution:showRelatedIntegrations`.
```
Comment (benironside) [inline docs/detections/rules-ui-manage.asciidoc:102]: maybe do "related integrations" with a capital at the beginning, or in code text

File: docs/getting-started/advanced-setting.asciidoc [hunk 5]
```diff
@@ -161,3 +162,9 @@ Adds a link to \https://www.dnschecker.org on *IP detail* pages:
 Each time a detection rule runs using a remote cross-cluster search (CCS) index pattern, it will return a warning saying that the rule may not have the required `read` privileges to the remote index. Because privileges cannot be checked across remote indices, this warning displays even when the rule actually does have `read` privileges to the remote index.
 
 If you've ensured that your detection rules have the required privileges across your remote indices, you can use the `securitySolution:enableCcsWarning` setting to disable this warning and reduce noise.
+
+[discrete]
+[[show-related-integrations]]
+== Show/hide related integrations in Rules page tables
+
+By default, Elastic prebuilt rules in the *Rules* and *Rule Monitoring* tables include a badge showing how many related integrations have been installed. Turn off `securitySolution:showRelatedIntegrations` to hide this in the rules tables (related integrations will still appear on rule details pages).
```
Comment (benironside) [inline docs/getting-started/advanced-setting.asciidoc:170]: maybe link to advanced settings here?

**Review by benironside — APPROVED** [review]
Looks great, left two minor suggestions

File: docs/detections/rules-ui-manage.asciidoc (hunk 4 above)
Comment (joepeeples) [reply docs/detections/rules-ui-manage.asciidoc:102]: I think just "integrations" to match the UI text, and bold because it's a UI label the user clicks, not text they enter.

I'll change this in a separate comment, so I can include the reference to the badge in line 97 above too.

Comment (joepeeples) [inline docs/detections/rules-ui-manage.asciidoc:102]:
```suggestion
You can also check rules' related integrations in the *Rules* and *Rule Monitoring* tables. Click the *integrations* badge to display the related integrations in a popup.

[role="screenshot"]
image::images/rules-table-related-integrations.png[Rules table with related integrations popup,75%]

TIP: You can hide the *integrations* badge in the Rules tables. Go to *{kib}* -> *Stack Management* -> *Advanced Settings*, then turn off `securitySolution:showRelatedIntegrations`.
```
